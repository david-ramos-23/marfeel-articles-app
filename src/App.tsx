import 'normalize.css';
import React from 'react';

export class App extends React.PureComponent {
  public render(): JSX.Element {
    return <div>My App</div>;
  }
}

export default App;
