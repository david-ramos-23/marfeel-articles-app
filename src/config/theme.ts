import logo from '../../assets/img/logo-marfeel.svg';
import { ThemeProps } from './types';

export const theme: ThemeProps = {
  colors: {
    primary: 'black',
    secondary: 'white',
  },
  sizes: {
    size1: '10px',
    size2: '15px',
    size3: '20px',
    size4: '40px',
    size5: '50px',
  },
  sideMenu: {
    headerBackground: 'rgb(2,0,36)',
    bodyBackground: 'white',
    linkColor: '#b9b9b9',
    linkColorActive: '#de0000',
    fontFamily: 'Roboto',
    fontWeight: '400',
    fontSize: '16px',
    items: [
      'HOME',
      'MAIN SECTION 1',
      'MAIN SECTION 2',
      'MAIN SECTION 3',
      'MAIN SECTION 4',
      'MAIN SECTION 5',
      'MAIN SECTION 6',
      'MAIN SECTION 7',
      'MAIN SECTION 8',
      'MAIN SECTION 9',
    ],
  },
  header: {
    background: 'rgb(0,0,0)',
    fontFamily: 'Roboto',
  },
  footer: {
    background: '#f2f2f2',
    textColor: '#888',
    fontFamily: 'Open Sans',
    fontWeight: '200',
    fontSize: '12px',
  },
};
