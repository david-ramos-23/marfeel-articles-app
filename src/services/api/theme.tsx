import request from '../request';

export const getTheme = async () => request.get(`theme`);
