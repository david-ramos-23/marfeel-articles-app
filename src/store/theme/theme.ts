import { ThemeState } from './types';
import logo from './../../assets/img/logo-marfeel.svg';

export const defaultTheme: ThemeState = {
  colors: {
    primary: 'black',
    secondary: 'white',
    gray1: '#E0E0E0',
    gray2: '#f2f2f2',
    gray3: '#b9b9b9',
  },
  sizes: {
    size1: '10px',
    size2: '15px',
    size3: '20px',
    size4: '40px',
    size5: '50px',
  },
  sideMenu: {
    headerBackground: 'rgb(0,0,0)',
    bodyBackground: 'white',
    linkColor: '#b9b9b9',
    linkColorActive: '#de0000',
    fontFamily: 'Roboto',
    fontWeight: '400',
    fontSize: '16px',
    items: ['HOME', 'TRAVELS', 'TECH', 'NATIONAL', 'INTERNATIONAL'],
  },
  header: {
    background: 'rgb(0,0,0)',
    fontFamily: 'Roboto',
    logo: {
      width: '100px',
      height: '25px',
      src: logo,
      title: 'Logo',
    },
    section: {
      fontWeight: '400',
      fontSize: '14px',
      borderWidth: '2px',
      itemPadding: '2px',
      colour: 'white',
      items: ['HOME', 'TRAVELS', 'TECH', 'NATIONAL', 'INTERNATIONAL'],
    },
    menu: {
      color: 'white',
      background: 'black',
    },
  },
  articles: {
    fontFamily: 'Open Sans',
  },
  footer: {
    background: '#f2f2f2',
    textColor: '#888',
    fontFamily: 'Open Sans',
    fontWeight: '200',
    fontSize: '12px',
  },
};
