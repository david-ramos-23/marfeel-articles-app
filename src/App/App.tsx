import 'normalize.css';
import React from 'react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../config/theme';
import { Main } from '../components/Main/Main';
import { Header } from '../components/Header';
import { Footer } from '../components/Footer';

export class App extends React.PureComponent {
  public render(): JSX.Element {
    return (
      <ThemeProvider theme={theme}>
        <>
          <Header />
          <Main />
          <Footer />
        </>
      </ThemeProvider>
    );
  }
}

export default App;
