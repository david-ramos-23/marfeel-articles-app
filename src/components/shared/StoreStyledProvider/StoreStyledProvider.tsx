import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import configureStore from '../../../store';
import { defaultTheme } from '../../../store/theme/theme';
import { SectionProvider } from '../../../hooks/useSection';

export const StoreStyledProvider: React.FunctionComponent = ({ children }) => (
  <Provider store={configureStore()}>
    <ThemeProvider theme={defaultTheme}>
      <SectionProvider>
        <>{children}</>
      </SectionProvider>
    </ThemeProvider>
  </Provider>
);
