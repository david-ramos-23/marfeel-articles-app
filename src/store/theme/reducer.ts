import { LOAD_THEME, ThemeActionTypes, ThemeState } from './types';
import { defaultTheme } from './theme';

const initialState: ThemeState = {
  ...defaultTheme,
};

export const themeReducer = (
  state = initialState,
  action: ThemeActionTypes
): ThemeState => {
  switch (action.type) {
    case LOAD_THEME: {
      return {
        ...state,
        ...action.theme,
      };
    }
    default:
      return state;
  }
};
