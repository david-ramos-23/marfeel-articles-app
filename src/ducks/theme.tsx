import { combineReducers } from 'redux';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { AnyAction } from 'redux';
import { Theme } from '../types/Theme';
import { ThemeState } from '../store/theme/types';
import { getTheme } from '../services/api/theme';
import logo from './../../assets/img/logo-marfeel.svg';

const GET_THEME_STARTED = 'GET_THEME_STARTED';
const GET_THEME_SUCCEEDED = 'GET_THEME_SUCCEEDED';
const GET_THEME_FAILED = 'GET_THEME_FAILED';

export interface State {
  theme: Theme;
}

export function getThemeThunk(): ThunkAction<Promise<void>, {}, {}, AnyAction> {
  return async (dispatch: ThunkDispatch<{}, {}, AnyAction>): Promise<any> => {
    dispatch(getThemeThunkStarted());
    try {
      const result = await getTheme();
      dispatch(getThemeThunkSucceeded(result));
      return result;
    } catch (exception) {
      console.log(exception);
      dispatch(getThemeThunkFailed());
      throw exception;
    }
  };
}

function getThemeThunkStarted() {
  return {
    type: GET_THEME_STARTED,
  };
}

function getThemeThunkSucceeded(result: Theme) {
  return {
    type: GET_THEME_SUCCEEDED,
    ...result,
  };
}

function getThemeThunkFailed() {
  return {
    type: GET_THEME_FAILED,
  };
}

const initialThemeState = {
  colors: {
    primary: 'black',
    secondary: 'white',
    gray1: '#E0E0E0',
    gray2: '#f2f2f2',
    gray3: '#b9b9b9',
  },
  sizes: {
    size1: '10px',
    size2: '15px',
    size3: '20px',
    size4: '40px',
    size5: '50px',
  },
  sideMenu: {
    headerBackground: 'rgb(0,0,0)',
    bodyBackground: 'white',
    linkColor: '#b9b9b9',
    linkColorActive: '#de0000',
    fontFamily: 'Roboto',
    fontWeight: '400',
    fontSize: '16px',
    items: [
      'HOME',
      'MAIN SECTION 1',
      'MAIN SECTION 2',
      'MAIN SECTION 3',
      'MAIN SECTION 4',
      'MAIN SECTION 5',
      'MAIN SECTION 6',
      'MAIN SECTION 7',
      'MAIN SECTION 8',
      'MAIN SECTION 9',
    ],
  },
  header: {
    background: 'rgb(0,0,0)',
    fontFamily: 'Roboto',
    logo: {
      width: '100px',
      height: '25px',
      src: logo,
      title: 'Logo',
    },
    section: {
      fontWeight: '400',
      fontSize: '14px',
      borderWidth: '2px',
      itemPadding: '2px',
      colour: 'white',
      items: [
        'SECTION 1',
        'SECTION 2',
        'SECTION 3',
        'SECTION 4',
        'SECTION 5',
        'SECTION 6',
        'SECTION 7',
        'SECTION 8',
        'SECTION 9',
        'SECTION 10',
      ],
    },
    menu: {
      color: 'white',
      background: 'linear-gradient(to bottom, #fd8d00 0%,#98cd00 100%)',
    },
  },
  footer: {
    background: '#f2f2f2',
    textColor: '#888',
    fontFamily: 'Open Sans',
    fontWeight: '200',
    fontSize: '12px',
  },
};

const theme = (state: ThemeState = initialThemeState, action: AnyAction) => {
  switch (action.type) {
    case GET_THEME_SUCCEEDED:
      return state;
    default:
      return state;
  }
};

export default combineReducers({
  theme,
});

export const getThemeState = (state: State) => state.theme;
