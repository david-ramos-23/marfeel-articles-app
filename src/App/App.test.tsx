import React from 'react';
import renderer from 'react-test-renderer';

import { App } from './App';

import { StoreStyledProvider } from '../components/shared/StoreStyledProvider';
import { defaultTheme } from '../store/theme/theme';
import { shallow } from 'enzyme';

describe('App', () => {
  const props = {
    theme: defaultTheme,
    loadTheme: jest.fn(),
  };
  let wrapper;

  it('renders without crashing', () => {
    const header = renderer
      .create(
        <StoreStyledProvider>
          <App {...props} />
        </StoreStyledProvider>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  it('loads the theme on componentDidMount', () => {
    wrapper = shallow(<App {...props} />);
    expect(props.loadTheme).toHaveBeenCalled();
  });
});
