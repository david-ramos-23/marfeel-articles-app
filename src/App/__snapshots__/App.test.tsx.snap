// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
Array [
  <header
    className="sc-fzqBZW gIrmBJ"
  >
    <div />
    <section
      className="sc-fzqNJr bTRQDu"
    >
      <button
        className="sc-fzoLsD MgXfQ"
        onClick={[Function]}
      >
        <div
          className="sc-fzpans YIAQv"
        />
      </button>
      <div
        className="sc-fzoyAV fRWUNs"
      >
        <img
          alt="Logo"
          className="sc-fzozJi gNayaJ"
          src="logo-marfeel.svg"
        />
      </div>
    </section>
    <nav
      className="sc-fznyAO bxLxaK"
    >
      <ul
        className="sc-fznKkj kHquXd"
      >
        <li
          className="sc-fznZeY gwNUTe"
        >
          <a
            className="sc-fzokOt fOaGtf"
            href="#1"
            onClick={[Function]}
            selected={true}
          >
            HOME
          </a>
        </li>
        <li
          className="sc-fznZeY gwNUTe"
        >
          <a
            className="sc-fzokOt ctaBng"
            href="#2"
            onClick={[Function]}
            selected={false}
          >
            TRAVELS
          </a>
        </li>
        <li
          className="sc-fznZeY gwNUTe"
        >
          <a
            className="sc-fzokOt ctaBng"
            href="#3"
            onClick={[Function]}
            selected={false}
          >
            TECH
          </a>
        </li>
        <li
          className="sc-fznZeY gwNUTe"
        >
          <a
            className="sc-fzokOt ctaBng"
            href="#4"
            onClick={[Function]}
            selected={false}
          >
            NATIONAL
          </a>
        </li>
        <li
          className="sc-fznZeY gwNUTe"
        >
          <a
            className="sc-fzokOt ctaBng"
            href="#5"
            onClick={[Function]}
            selected={false}
          >
            INTERNATIONAL
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <section
    className="sc-AxheI dRPfAF"
  />,
  <footer
    className="sc-Axmtr krLLiE"
  >
    <div
      className="sc-AxmLO hgdTff"
    >
      ï¿½Copyright 2020 - marfeel.com
    </div>
  </footer>,
  <div
    className="sc-fzoLag kzFhbn"
    onClick={[Function]}
  />,
  <div
    className="sc-fzoXzr eOeUjM"
  >
    <header
      className="sc-fzpjYC dALNtZ"
    >
      <img
        alt="Logo"
        className="sc-fzozJi gNayaJ"
        src="logo-marfeel.svg"
      />
    </header>
    <nav
      className="sc-fznxsB epyCZq"
    >
      <ul
        className="sc-fznJRM jzZaHF"
      >
        <li
          className="sc-fznWqX haNDGY"
        >
          <a
            className="sc-fzoiQi kAOSFW"
            href="#1"
            onClick={[Function]}
          >
            HOME
          </a>
        </li>
        <li
          className="sc-fznWqX haNDGY"
        >
          <a
            className="sc-fzoiQi eBoAcE"
            href="#2"
            onClick={[Function]}
          >
            TRAVELS
          </a>
        </li>
        <li
          className="sc-fznWqX haNDGY"
        >
          <a
            className="sc-fzoiQi eBoAcE"
            href="#3"
            onClick={[Function]}
          >
            TECH
          </a>
        </li>
        <li
          className="sc-fznWqX haNDGY"
        >
          <a
            className="sc-fzoiQi eBoAcE"
            href="#4"
            onClick={[Function]}
          >
            NATIONAL
          </a>
        </li>
        <li
          className="sc-fznWqX haNDGY"
        >
          <a
            className="sc-fzoiQi eBoAcE"
            href="#5"
            onClick={[Function]}
          >
            INTERNATIONAL
          </a>
        </li>
      </ul>
    </nav>
  </div>,
]
`;
